tests/integration/server/courseInstance/courseInstance.service.test.ts

```diff
@@ -131,6 +131,31 @@ describe('Course Instance Service', function () {
         });
       });
     });
+    it('Should list the faculty notes if any for each instance', async function () {
+      const dbInstances = await instanceRepository.find({
+        relations: ['facultyCourseInstances', 'facultyCourseInstances.faculty'],
+      });
+      notStrictEqual(result.length, 0);
+      result.forEach(({ spring, fall }) => {
+        [spring, fall].forEach(({ id, instructors, offered }) => {
+          strictEqual(Array.isArray(instructors), true);
```

Comment:
That's a good point. I was actually mirroring my test with the test above it, which I think Jonathan wrote (line 108) - "Should list the instructors for each offered instance in the correct order," and it checks each item in the loop.

I added error messages for checking that the `result` length is not `0` and to check that the `instructors` is an array. I think the purpose of these tests is so that we don't try to loop if there is no length and also if it's not an array. I'm not sure if it would make sense to check whether the instructors is an array outside of the test, but does it help to have added these error messages?
![image](https://user-images.githubusercontent.com/29589689/112506194-e3ea6480-8d63-11eb-9aea-7c4b515a5266.png)

This is the test that I mirrored to write mine: 
![image](https://user-images.githubusercontent.com/29589689/112506440-21e78880-8d64-11eb-9705-c12168b9d7a4.png)

Reply:
So, the first `notStrictEqual` is fine, it's there to fail the test if the array has 0 elements. Otherwise, the `forEach` would run 0 times and the test would show up as having passed without actually asserting anything(this is probably an instance where a richer assertion library might be handy to say something like `expect.assertions(3)` or something - but that's a slight side-issue).

What I was really looking at here were the other two - 

- The first one is checking that `instructors` is an array. I would say this isn't strictly necessary since if `instructors` is `null`, `undefined`, an `object` or anything else - an error would be thrown anyway since none of those have a `.forEach` method. The only exception would be if `instructors` turned out to be a `Map` (which _does_ have a `.foreEach` method)....but again that starts to get into the weeds of testing the type system and stray away from what the test claims to be testing. Personally - I would just take this out (or move it to another test)...

- The second one is testing that the notes are what's expected for each instructor. I was going to suggest flattening the arrays and comparing them, I ended up with was something like this....
```js
const instructors = dbInstances.map(
  ({ facultyCourseInstances }) => facultyCourseInstances
  .reduce((acc, instance) => acc.concat(instance.faculty), [])
).reduce((acc, val) => acc.concat(val), []);

const notes = result.map(({ spring, fall }) => [spring, fall])
  .reduce((acc, val) => acc.concat(val), [])
  .filter(({ offered }) => offered === OFFERED.Y)
  .map(({ instructors }) =>
    instructors.reduce((notesArray, { id, notes }) => notesArray.concat({ id, notes }), [])
  )
  .reduce((acc, val) => acc.concat(val), []);

notes.forEach(({ id, notes }) => {
  const instructor = instructors
    .find(({ id: instructorId }) => instructorId === id);

  deepStrictEqual([id, notes], [instructor.id, instructor.notes]);
});
```
This works, but, I'm not sure it's really better - so feel free to disregard this point if you want :)

...in fact, if anything - looking at it again, I'd argue that this is actually _less_ clear than what was there...